docs/_includes/Refinement.md

```diff
@@ -0,0 +1,130 @@
+
+0) The refinement result is a module within the same enclosing module as the
+refining module, has the same name, and in fact replaces the refining module  in their shared scope.
+
+1) All the declarations (including import and export declarations) of the parent are copied into the refinement result.
+These declarations are _not_ re-resolved. That is, the assignment of 
+declarations and types to syntactic names is not changed. The refinement 
+result may exist in a different enclosing module and with a different set of
+imports than the refinement parent, so that if names were reresolved, the 
+result might be different (and possibly not semantically valid).
+Hence it is important that the names not be re-resolved in their new context.
+
+2) All the declarations of the refining module that have different names
+than the declarations in the refinement parent are also copied into the 
+refinement result. Again name and type resolution of the refining module
+happen before this combination takes place.
+
+3) Where declarations in the parent and refinement module have the same name,
+the second refines the first and the combination, a refined declaration, is
+the result placed in the refinement result module, to the exclusion of the
+declarations with the same name from the parent and refinement modules.
+
+The way declarations are combined depends on the kind of declaration;
+the combination rules are described in subsections below.
+
+So that it is clear that refinment is taking place, refining declarations
+have some syntactic indicator that they are refining some parent declaration.
+Typically this is the presence of a `...` token.
+
+## 22.47. Export set declarations
+
+A refining export set declaration begins with the syntax
+```grammar
+"export" Ident "..." 
+```
+but otherwise contains the same `provides`, `reveals` and `extends` sections,
+with the ellipsis indicating that it is a refining declaration.
+
+The combined result declaration has the same name as the two input declarations and the unions of names from each of the `provides`, `reveals`, and `extends`
+sections, respectively.
+
+An unnamed export set declaration from the parent is copied into the result
+module with the name of the parent module. The result module has a default
+export set according to the general rules for export sets, after all of 
+the result module's export set declarations have been assembled.
+
+## 22.48. Import declarations
+
+Aliasing import declarations are not refined. The result module contains the union
+of the import declarations from the two input modules.
+There must be no names in common among them.
+
+Abstract import declarations (declared with `:` instead of `=`, [Section 0](#sec-module-abstraction)) are refined. The refinement parent contains the 
+abstract import and the refining module contains a regular aliasing
+import for the same name. Dafny checks that the refining import _adheres_ to
+the abstract import.
+
+TODO: Add example?
```

Comment:
I wouldn't bother with an example at this time, especially since we'll be working to replace abstract import declarations with module parameters.

Reply:
OK